app_dart/test/request_handlers/refresh_github_commits_test.dart

```diff
@@ -173,6 +178,18 @@ void main() {
       expect(tester.log.records.where(hasLevel(LogLevel.ERROR)), isEmpty);
     });
 
+    test('inserts the latest single commit if a new branch is found', () async {
+      githubCommits = <String>['1', '2', '3', '4', '5', '6', '7', '8', '9'];
+      githubBranches = <String>['flutter-0.0-candidate.0'];
+      config.newBranchHoursValue = 168;
+
+      expect(db.values.values.whereType<Commit>().length, 0);
+      httpClient.request.response.body = singleTaskManifestYaml;
+      branchHttpClient.request.response.body = branchRegExp;
+      await tester.get<Body>(handler);
+      expect(db.values.values.whereType<Commit>().length, 1);
+    });
+
```

Comment:
What happens in the case that a hotfix branch is created and it has several commits cherry picked to it?

Reply:
My understanding is that the release branch is created from master first (this case it will check the latest commit only), and then cherry picks happen later (this case we will check all commits picked up).

Does the case Casey mentioned above happen @tvolkert ?

Reply:
cc @godofredoc

Reply:
Added some comments along the different files